spacy/kb_base.pxd

```diff
@@ -0,0 +1,25 @@
+"""Knowledge-base for entity or concept linking."""
+
+from cymem.cymem cimport Pool
+from libcpp.vector cimport vector
+from libc.stdint cimport int64_t
+
+from .vocab cimport Vocab
+from .typedefs cimport hash_t
+from .kb_base cimport BaseKnowledgeBase
+
+
+# Object used by the Entity Linker that summarizes one entity-alias candidate combination.
+cdef class Candidate:
+    cdef readonly BaseKnowledgeBase kb
+    cdef hash_t entity_hash
+    cdef float entity_freq
+    cdef vector[float] entity_vector
+    cdef hash_t alias_hash
+    cdef float prior_prob
+
+
+cdef class BaseKnowledgeBase:
```

Comment:
It irks me that I couldn't figure out how to mark this class as abstract. Is there really no way to do so in Cython?

Reply:
I don't think so. We have a similar situation with `cdef class Pipe` - basically it's "abstract" by raising `NotImplementedError`'s if you'd use it directly...